clang/lib/Frontend/InitPreprocessor.cpp

```diff
@@ -113,7 +113,11 @@ static T PickFP(const llvm::fltSemantics *Sem, T IEEEHalfVal, T IEEESingleVal,
 
 static void DefineFloatMacros(MacroBuilder &Builder, StringRef Prefix,
                               const llvm::fltSemantics *Sem, StringRef Ext) {
-  const char *DenormMin, *Epsilon, *Max, *Min;
+  const char *DenormMin, *NormMax, *Epsilon, *Max, *Min;
+  NormMax = PickFP(Sem, "6.5504e+4", "3.40282347e+38",
+                   "1.7976931348623157e+308", "1.18973149535723176502e+4932",
+                   "1.79769313486231580793728971405301e+308",
```

Comment:
I don't think this is the correct value for ppc double-double, since double-double is supposed to give a different result for NORM_MAX than MAX (indeed, that's explicitly the point of the new macros!)

Reply:
Good catch, thank you! I see now that GCC uses the following values for `long double`:
```
#define __LDBL_MAX__ 1.79769313486231580793728971405301e+308L
#define __LDBL_NORM_MAX__ 8.98846567431157953864652595394501e+307L
```
so I should probably go with `8.98846567431157953864652595394501e+307` as the value? CC @hubert-reinterpretcast

Reply:
Ping -- I think I've corrected the value (at least, I'm using the same value as GCC in this case).

Reply:
I'm pretty sure this is correct, but I'd rather prefer someone who knows PPC to confirm this.

Reply:
Ping @hubert-reinterpretcast @chenzheng1030 -- hoping to get this landed for Clang 19, just need verification that the value is correct here.

Reply:
Yes, the value is correct for an interpretation where the `LDBL_MANT_DIG` is 106.

Reply:
Thank you Hubert!